### PR TITLE
delete action implemented

### PR DESCRIPTION
**Pull request by peratik**
## Changes
- Delete action implemented
- Confirm delete dialog added
- Loading state added

## Checklist

- [x] I've performed a self-review, followed all rules from [Frontend style guide](https://www.notion.so/santiment/Front-end-style-guide-81750096b38c4bea9a29b14fd4ab8667)
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've updated the [documentation](https://github.com/santiment/academy), if necessary (Keyboard shortcuts, Account settings)
- [x] I've checked night mode, mobile & tablet screens (if have changes in UI)
- [x] I've added tests (if necessary)

### CODE REVIEW

File: src/pages/Explorer/Components/api/delete.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    `
+        mutation {
+            remove: deleteChartConfiguration(id: ${id}) {
+                id 
+            }
+        }
+    `,
+}
+
+export default (id, deleteKey) => mutate(DELETE[deleteKey](id))
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/api/delete.js:30]:
```suggestion
function DELETE_ENTITY_MUTATION(id, type) {
  let data = 'id'
  let mutation = 'removeWatchlist'

  switch (type) {
    case Something.CHART:
      mutation = 'deleteChartConfiguration'
      break
    case Something.TRIGGER:
      mutation = 'removeTrigger'
      data = 'trigger { id }'
  }

  return `mutation {
    ${mutation}(id:${id}){
      ${data}
    }
  }`
}

export const mutateDeleteEntity = (id, type) => mutate(DELETE_ENTITY_MUTATION(id, type))
```

File: src/pages/Explorer/Components/DeleteConfirmationDialog.svelte [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    <div class="column hv-center body-2 c-waterloo mrg--b mrg-xl">
+      Are you sure? This action cannot be undone.
+    </div>
+    <div class="row hv-center">
+      <button
+        class="btn-1 mrg--r mrg-m"
+        on:click={() =>
+          onDelete()
+            .then((id) => onComplete(id))
+            .then(closeDialog)}>Delete</button
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/DeleteConfirmationDialog.svelte:34]:
```suggestion
        on:click={onDeleteClick}>Delete</button
```

File: src/pages/Explorer/Components/DeleteConfirmationDialog.svelte [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  $: ({ itemType, onDelete, onComplete } = $$props)
+</script>
+
+<Dialog {...$$props} noTitle bind:closeDialog>
+  <div class="dialog">
+    <div class="row justify v-center h4 txt-m mrg--b mrg-s c-black">
+      Do you want to delete this {itemType}?
+      <Svg id="close" w="12" on:click={closeDialog} class="btn" />
+    </div>
+    <div class="column hv-center body-2 c-waterloo mrg--b mrg-xl">
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/DeleteConfirmationDialog.svelte:25]: 1. `column` is redundant;
2. `[D-0-1]` Change the order of the margin classes: 1) `mrg-[size]` 2) `mrg-[side]`

File: src/pages/Explorer/Components/DeleteConfirmationDialog.svelte [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  import Svg from 'webkit/ui/Svg/svelte'
+
+  let closeDialog
+
+  $: ({ itemType, onDelete, onComplete } = $$props)
+</script>
+
+<Dialog {...$$props} noTitle bind:closeDialog>
+  <div class="dialog">
+    <div class="row justify v-center h4 txt-m mrg--b mrg-s c-black">
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/DeleteConfirmationDialog.svelte:21]: Same as the `[D-0-1]`

File: src/pages/Explorer/Components/DeleteConfirmationDialog.svelte [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      Do you want to delete this {itemType}?
+      <Svg id="close" w="12" on:click={closeDialog} class="btn" />
+    </div>
+    <div class="column hv-center body-2 c-waterloo mrg--b mrg-xl">
+      Are you sure? This action cannot be undone.
+    </div>
+    <div class="row hv-center">
+      <button
+        class="btn-1 mrg--r mrg-m"
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/DeleteConfirmationDialog.svelte:30]: Same as the `[D-0-1]`

File: src/pages/Explorer/Components/DeleteConfirmationDialog.svelte [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        on:click={() =>
+          onDelete()
+            .then((id) => onComplete(id))
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/DeleteConfirmationDialog.svelte:33]: `.then(onComplete)`

File: src/pages/Explorer/Components/Actions.svelte [hunk 7]
```diff
@@ -56,16 +58,25 @@
 
     history.push(getItemRoute(item, type), { openComments: true })
   }
+
+  function onDelete(e) {
+    e.preventDefault()
+    showDeleteConfirmationDialog(
+      singular,
+      () => deleteAction(id, deleteKey),
+      () => document.getElementById(`${getItemId(item, type)}`).parentElement.remove(),
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/Actions.svelte:67]: Filter the array that contains this item.

File: src/pages/Explorer/Components/Actions.svelte [hunk 8]
```diff
@@ -56,16 +58,25 @@
 
     history.push(getItemRoute(item, type), { openComments: true })
   }
+
+  function onDelete(e) {
+    e.preventDefault()
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/Actions.svelte:63]: `[D-0-2]` Use the [`|preventDefault`](https://svelte.dev/docs#template-syntax-element-directives-on-eventname) modifier

File: src/pages/Explorer/Components/Actions.svelte [hunk 9]
```diff
@@ -20,7 +21,11 @@
   let label = ''
   let voteTimeout
 
-  function onShare() {
+  $: id = item.trigger ? item.trigger.id : item.id
+  $: ({ voteKey, deleteKey, singular } = EntityType[type])
+
+  function onShare(e) {
+    e.preventDefault()
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Components/Actions.svelte:28]: Same as the `[D-0-2]`

File: src/pages/Explorer/Category/UserCreation.svelte [hunk 10]
```diff
@@ -18,14 +18,18 @@
   $: ({ user, commentsCount } = item)
 
   function onClick(e) {
+    if (['use', 'svg'].indexOf(e.target.tagName) > -1) {
+      e.preventDefault()
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Category/UserCreation.svelte:22]: Same as the `[D-0-2]`

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Explorer/api.js [hunk 11]
```diff
@@ -57,7 +57,7 @@ export const queryExplorerItems = ({
         }
       }
     `
-  return query(QUERY).then(accessor)
+  return query(QUERY, { cache: false }).then(accessor)
```
[the remark this answers is not in the document]
Comment (peratik) [reply src/pages/Explorer/api.js:60]: Is 20 sec good?

File: src/pages/Explorer/Category/UserCreation.svelte [hunk 12]
```diff
@@ -18,14 +18,18 @@
   $: ({ user, commentsCount } = item)
 
   function onClick(e) {
+    if (['use', 'svg'].indexOf(e.target.tagName) > -1) {
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/Category/UserCreation.svelte:21]:
```suggestion
    if (['use', 'svg'].includes(e.target.tagName)) {
```

File: src/pages/Explorer/api.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (DmitriVanGuard) [reply src/pages/Explorer/api.js]: Yes

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Explorer/index.svelte [hunk 13]
```diff
@@ -10,51 +10,70 @@
...
+    min-height: 100%;
+  }
+
+  .aside {
+    width: 320px;
+    margin-left: 48px;
+    position: sticky;
+    bottom: 0;
+    align-self: flex-end;
   }
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/index.svelte:77]:
```css
main, .aside {
   position: sticky;
   bottom: 0;
   align-self: flex-end;
}
```

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Explorer/index.svelte [hunk 14]
```diff
@@ -10,51 +10,71 @@
...
+  main,
+  .aside {
+    position: sticky;
+    bottom: 0;
+    align-self: flex-end;
+  }
+
   main {
     width: 640px;
+    min-height: 100%;
```
Comment (DmitriVanGuard) [inline src/pages/Explorer/index.svelte:72]:
```suggestion
```

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

**Pull request merged**